### ISSUE TITLE
Problem with switching to english language

### ISSUE THREAD

**Issue by RepositoryNY** [stefalda/react-localization#20]
Here my localization.js file:

```
import LocalizedStrings from 'react-localization';

const en = {
 ...
};

const uk = {
  ....
};

export const localization = new LocalizedStrings({ en, uk });

export const convertLanguage = function (language) {
  switch (language) {
    case 'uk':
      return { TwoLetter: 'uk', NativeName: 'Українська' };
    case 'en':
      return { TwoLetter: 'en', NativeName: 'English' };
    default:
      return { TwoLetter: 'df', NativeName: 'Default' };
  }
}

export const extractAvailableLanguages = function (locs) {
  return locs.getAvailableLanguages().filter(x => x !== locs.getLanguage());
}

```

I'm using it in App.js file:

```
import React, { Component } from 'react';
import MainPage from './MainPage';
import { localization as _l, convertLanguage } from '../localization';
export default class App extends Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    $('.dropdown-button').dropdown();
  }
  setLanguage(twoLetter, e) {
    e.preventDefault();
    _l.setLanguage(twoLetter);
    this.forceUpdate();
  }
  render() {
    const currentLanguage = convertLanguage(_l.getLanguage());
    return (
      <div>
        <ul id='languageDropdown' className='dropdown-content'>
          {_l.getAvailableLanguages().map(item => convertLanguage(item)).map((item, i) =>
            <li key={i}>
              <a href='' onClick={this.setLanguage.bind(this, item.TwoLetter)}>
                {item.NativeName}
              </a>
            </li>
          )}
        </ul>
        <nav className='cyan'>
          <div className='nav-wrapper'>
            <a className='brand-logo' style={{ marginLeft: 16 }}>MyApp</a>
            <ul className='right hide-on-med-and-down'>
              <li>
                <a className='dropdown-button' href='' data-activates='languageDropdown'>
                  {currentLanguage.NativeName}
                  <i className='material-icons right'>arrow_drop_down</i>
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <div className='container'>
          <MainPage />
        </div>
        <footer className='page-footer cyan'>
          <div className='footer-copyright'>
            <div className='container'>
              <span style={{ fontWeight: 'normal' }}>© 2017 MyApp</span>
            </div>
          </div>
        </footer>
      </div>
    )
  }
}
```

Here, I have two problems:

1.  Switching to ukrainian language works good, but child components in `MainPage` doesn't rerendering. How can I rerender all my app, to switch language dynamically?
2. When I'm switching to english language, nothing happens. It seems like it happens only with english language, because with other languages it works pretty fine.

So, how can I solve these problems?

**Comment by andy-pro** [comment]
I have the same problem!
Problem is in line 54 of source file [LocalizedStrings](https://github.com/stefalda/react-localization/blob/master/src/LocalizedStrings.js)

    var localizedStrings = Object.assign(this.props[Object.keys(this.props)[0]], this.props[this.language]);

As a result, any selected language overwrites the element with index 0 of the array of languages (Object.keys).
I solved the problem in such a way:

    let strings = new LocalizedStrings({
        default: {},
        en: {     ...       },
        ru: {     ...       },
        ua: {     ...       },
    });

and then set default language to 'en'.
Properties order in objects is not guaranteed in JavaScript, but it's work for me

**Comment by stefalda** [comment]
If you change line 54 back to its original implementation
`
 var localizedStrings = this.props[this.language];`

it works as expected.

I'll have to refactor that code that has been added to preserve missing strings.

**Comment by stefalda** [comment]
I've just released version 0.12 that should fix the problem
